### PR TITLE
Allow overriding eth.radicle.seed.host record

### PR DESCRIPTION
**Pull request by cloudhead**
Adds two new records:

* eth.radicle.seed.api
* eth.radicle.seed.git

When specified, these override `eth.radicle.seed.host` for Git
and the HTTP API, respectively.

### CODE REVIEW

File: src/profile.ts
Comment (sebastinez) [inline src/profile.ts:144]: I think this is a good place to use the nullish coalescing operator, since we are checking for nullish values.

```suggestion
        return { address, idx: idx ?? undefined };
```

File: src/config.ts
Comment (sebastinez) [inline src/config.ts:148]: I think this is a good place to use the nullish coalescing operator, since we are checking for nullish values.

```suggestion
    const api = seed.api ?? seed.host;
    const git = seed.git ?? seed.host;
```

**Review by sebastinez — APPROVED** [review]
LGTM, did just comment two details of using nullish coalescing operator instead of `||`. It works fine like it is, but I think while `??` checks for `null` or `undefined`, `||` has eventually a to wide scope

File: src/base/registrations/registrar.ts
Comment (sebastinez) [inline src/base/registrations/registrar.ts:92]: I think we have an issue here since `r.value` can also be `null`, since `getText` returns `string | null`. Eventually we could do something like 
```suggestion
    meta.map(r => r.status == "fulfilled" ? r.value ?? undefined : undefined);
```

File: src/base/registrations/registrar.ts
Comment (sebastinez) [inline src/base/registrations/registrar.ts:114]: Here we also will have an issue since `resolver.getText('avatar')` returns `Promise<string | null>` we return `null` and handle it later other await the promise and return `undefined`?

File: src/base/registrations/registrar.ts
Comment (sebastinez) [inline src/base/registrations/registrar.ts:124]: Here we also will have an issue since `resolver.getText('anchors')` returns `Promise<string | null>`

File: src/base/registrations/registrar.ts
Comment (sebastinez) [inline src/base/registrations/registrar.ts:134]: Same here returning `Promise<string | null>`

**Review by sebastinez — CHANGES_REQUESTED** [review]
I found some things that we should review before merging, since it will break the build.
All the functions that use `resolver.getText(something)` return `Promise<string | null>` but we want to return `Promise<string | undefined>`.
IMO it's nicer using `undefined` for various reasons. But to make it work, we would need to await the promises and handle the return null value or eventually create a PR on ethers.js to change this?

**Comment by cloudhead** [comment]
Thanks for the review!

So I think in terms of nullish values, the convention is that if you want to return "nothing", you return `null`, but if you want to take "nothing" as a parameter, you use `param?`, ie. you take `undefined`.

This is nice because it avoids accidently returning nothing when you wanted to return a value.

**Comment by sebastinez** [comment]
> So I think in terms of nullish values, the convention is that if you want to return "nothing", you return `null`, but if you want to take "nothing" as a parameter, you use `param?`, ie. you take `undefined`.

Yes this sounds nice and makes sense, but to comply with the types if you want to return "nothing" ie. `null` you would need to specify that as a return type instead of `undefined`.
IMO we should as you said in PR #115 we should agree on one nil type and use it everywhere instead of managing both.

**Comment by cloudhead** [comment]
I've fixed some of the null/undefined issues. Let me know if I missed anything!

File: src/base/registrations/registrar.ts
Comment (sebastinez) [inline src/base/registrations/registrar.ts:139]: In line with what we talked today, shouldn't we manage optional parameters only with `?` and pass only the required arguments to the functions after checking against `null`?

```suggestion
export async function getAnchorsAccount(name: string, config: Config, resolver?: EnsResolver): Promise<string | null> {
```

File: src/base/registrations/registrar.ts
Comment (sebastinez) [inline src/base/registrations/registrar.ts:149]:
```suggestion
export async function getSeed(name: string, config: Config, resolver?: EnsResolver): Promise<Seed | null> {
```

File: src/base/registrations/registrar.ts
Comment (sebastinez) [reply src/base/registrations/registrar.ts:139]: We would need to create some temporary value for the `config.provider.getResolver(name)` call though, since we won't be able to associate it to `resolver`, ie.

```ts
// using var here to shadow the function param
const newResolver = resolver ?? await config.provider.getResolver(name);
if (! newResolver) {
  return null;
}
```

File: src/base/registrations/registrar.ts
Comment (sebastinez) [inline src/base/registrations/registrar.ts:164]: This will be an issue, since `getText` could eventually return a `null` value, but in the creation of the `Seed` class we accept only `string | undefined`

Maybe we could do a map similar to.
```ts
const [host, id, git, api] =
    seedPromise.map(r => r.status == "fulfilled" && r.value ? r.value : undefined);
```

**Review by sebastinez — COMMENTED** [review]
LGTM, just left some details on managing `null` values

File: src/base/registrations/registrar.ts
Comment (cloudhead) [reply src/base/registrations/registrar.ts:164]: Hmm but what is the issue exactly? It seems like typescript is fine compiling this, and the code will work since we check for truthy-ness only?

Comment (cloudhead) [reply src/base/registrations/registrar.ts:164]: I guess you mean when we upgrade to ethers 6? I think we can just deal with that issue when/if the time comes..

File: src/base/registrations/registrar.ts
Comment (cloudhead) [reply src/base/registrations/registrar.ts:139]: Yeah, I wasn't going to do this in this PR, but let me check.

Comment (cloudhead) [reply src/base/registrations/registrar.ts:139]: Yeah I'd rather fix this separately.

File: src/base/registrations/registrar.ts
Comment (sebastinez) [reply src/base/registrations/registrar.ts:164]: Yeah I don't know why typescript compiles fine for you, for me it throws a typecheck error, that it can't assign a `string | null` value to `string | undefined`..
But if you want to go ahead, since the CI also doesn't throw errors and the build seems to work, why not :smiley:

**Review by sebastinez — APPROVED** [review]
If the typechecker doesn't throw errors I'm okay with it too :sweat_smile:

**Pull request merged**
